File: Backend/udp-server/src/main/java/net/chrisparton/sparkled/udpserver/UdpServer.java
package net.chrisparton.sparkled.udpserver;

import java.net.SocketException;

/**
 * A UDP Server that listens on the given port.
 */
public interface UdpServer {

    /**
     * Start the UDP server on the provided port. This method is idempotent.
     *
     * @param port The port to listen on.
     */
    void start(int port) throws SocketException;
}

File: udp-server/src/main/kotlin/io/sparkled/udpserver/UdpServer.kt
package io.sparkled.udpserver

/**
 * A UDP Server that listens on the given port.
 */
interface UdpServer {

    /**
     * Start the UDP server on the provided port. This method is idempotent.

     * @param port The port to listen on.
     */
    fun start(port: Int)
}
